.github/workflows/push.yml

```diff
@@ -14,8 +14,23 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ruby-version: [3.0, 2.7]
-        gemfile: ['gemfiles/nokogiri_13.10.gemfile', 'gemfiles/nokogiri_12.5.gemfile']
+        ruby-version: [3.4, 3.3, 3.2, 3.1, 3.0, 2.7]
+        gemfile: [
+          'gemfiles/nokogiri_1.12.gemfile',
+          'gemfiles/nokogiri_1.13.gemfile',
+          'gemfiles/nokogiri_1.14.gemfile',
+          'gemfiles/nokogiri_1.15.gemfile',
+          'gemfiles/nokogiri_1.16.gemfile',
+          'gemfiles/nokogiri_1.17.gemfile',
+          'gemfiles/nokogiri_1.18.gemfile',
+        ]
+        exclude:
+          - gemfile: 'gemfiles/nokogiri_1.16.gemfile' # requires ruby >= 3.0.0
+            ruby-version: 2.7
+          - gemfile: 'gemfiles/nokogiri_1.17.gemfile' # requires ruby >= 3.0.0
+            ruby-version: 2.7
+          - gemfile: 'gemfiles/nokogiri_1.18.gemfile' # requires ruby >= 3.1.0
+            ruby-version: [2.7, 3.0]
```

Comment:
you can't use an array to specify an exclude - each entry must be specified separately